src/main/java/com/paskar/email/application/configuration/HibernateConfig.java

```diff
@@ -0,0 +1,32 @@
+package com.paskar.email.application.configuration;
+
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.orm.jpa.JpaTransactionManager;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+
+import javax.persistence.EntityManagerFactory;
+import javax.sql.DataSource;
+
+@Configuration
+@EnableTransactionManagement
+@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
+@RequiredArgsConstructor
+public class HibernateConfig {
+
+    EntityManagerFactory factory;
+    DataSource dataSource;
+
+    @Bean
+    public PlatformTransactionManager transactionManager() {
```

Comment:
Inject Spring beans right into the method.
No need to make it as class field in this case.

Reply:
```
void PlatformTransactionManager(EntityManager em) {
    var tm = new JpaTransactionManager();
    tm.setEntityManager(em);
    return tm;
}
```